frontend/components/MostViewed.tsx

```diff
@@ -147,49 +148,199 @@ const headlineLink = css`
+                                                            ${screenReaderOnly};
+                                                        `}
+                                                    >
+                                                        Most viewed{' '}
+                                                    </span>
+                                                    {tab.heading}
+                                                </button>
+                                            </li>
+                                        ))}
+                                    </ol>
+                                )}
+                            {(data || []).map((tab, i) => (
+                                <ul
+                                    className={cx(list, {
+                                        [hideList]:
+                                            i !== this.state.selectedTabIndex,
+                                    })}
+                                    id={`tabs-popular-${i}`}
+                                    key={`tabs-popular-${i}`}
+                                    role="tabpanel"
+                                    aria-labelledby={`tabs-popular-${i}-tab`}
+                                >
+                                    {(tab.trails || []).map((trail, ii) => (
+                                        <li
+                                            className={listItem}
+                                            key={trail.url}
+                                        >
+                                            <span className={bigNumber}>
+                                                <BigNumber index={ii + 1} />
+                                            </span>
+                                            <h2 className={headlineHeader}>
+                                                <a
+                                                    className={headlineLink}
+                                                    href={trail.url}
+                                                >
+                                                    {trail.linkText}
+                                                </a>
+                                            </h2>
+                                        </li>
+                                    ))}
+                                </ul>
+                            ))}
+                        </div>
+                    )}
+                </AsyncClientComponent>
+            </div>
+        );
+    }
+
+    public fetchTrails: () => Promise<Tab[]> = () => {
+        const sectionsWithoutPopular = ['info', 'global'];
+        const hasSection =
+            this.props.sectionName &&
+            !sectionsWithoutPopular.includes(this.props.sectionName);
+        const endpoint = `/most-read${
+            hasSection ? `/${this.props.sectionName}` : ''
+        }.json`;
+        return new Promise((resolve, reject) => {
```

Comment:
A philosophical question. Does this logic belong here, at the presentational layer, or further up at the data layer in `parse-capi`? I'm not suggesting we move it, but we ought to start thinking about how much logic is executed down here among the presentational code

Reply:
If it were to exist in `parse-capi` would it not then be fetched from the server? The nature of most-read is that it's regularly changing? If it were in the server side rendered response it'd be cached and out of date, or we'd have to bust the cache everytime it updates. Also we want the initial render to be as fast as possible, which means constructing the `CAPI` model as quickly as we can, waiting for `async` calls to respond will slow things down.